websites/T/Tumblr/presence.ts

```diff
@@ -3,105 +3,178 @@ const presence = new Presence({
 	}),
 	browsingTimestamp = Math.floor(Date.now() / 1000);
 
-let user: HTMLElement, search: HTMLElement;
-
 presence.on("UpdateData", async () => {
 	const presenceData: PresenceData = {
-		largeImageKey:
-			"https://cdn.rcd.gg/PreMiD/websites/T/Tumblr/assets/logo.png",
-	};
+			largeImageKey:
+				"https://cdn.rcd.gg/PreMiD/websites/T/Tumblr/assets/logo.png",
+			startTimestamp: browsingTimestamp,
+		},
+		{ hostname, href, pathname } = document.location,
+		search = document.querySelector<HTMLInputElement>(
+			'[type="search"],#search_actions_search'
+		);
+
+	switch (hostname) {
+		case "www.tumblr.com": {
+			switch (true) {
+				case !!search?.value: {
+					presenceData.details = "Searching for";
+					presenceData.state = search.value;
+					presenceData.smallImageKey = Assets.Search;
+					break;
+				}
+				case !!pathname.match(/[0-9]{18}/gm)?.[0]: {
+					const author =
+							document.querySelector<HTMLAnchorElement>('[rel="author"]'),
+						video = document.querySelector<HTMLVideoElement>("video");
+
+					if (!isNaN(video?.duration)) {
+						delete presenceData.startTimestamp;
+						presenceData.smallImageKey = video.paused
+							? Assets.Pause
+							: Assets.Play;
+						presenceData.smallImageText = video.paused ? "Paused" : "Playing";
+						if (!video.paused) {
+							[, presenceData.endTimestamp] =
+								presence.getTimestampsfromMedia(video);
+						}
+						presenceData.buttons = author
+							? [
+									{ label: "Watch Video", url: href },
+									{
+										label: "View Creator's Profile",
+										url: author.href,
+									},
+							  ]
+							: [{ label: "Watch Video", url: href }];
+					} else {
+						presenceData.buttons = author
+							? [
+									{ label: "View Post", url: href },
+									{
+										label: "View Creator's Profile",
+										url: author.href,
+									},
+							  ]
+							: [{ label: "View Post", url: href }];
+					}
+					presenceData.details = document.querySelector(".k31gt");
+					presenceData.state = author?.textContent;
+
+					break;
+				}
+				case pathname === "/": {
+					presenceData.details = "Viewing the home page";
+					break;
+				}
+				case document.querySelector<HTMLMetaElement>('[property="og:type"]')
+					?.content === "profile": {
+					presenceData.details = `Viewing user: ${
+						document.querySelector(".vfPi2").textContent
```

Comment:
Try non-generated selectors if possible.
```suggestion
						document.querySelector("header h1").textContent
```

Reply:
> Try non-generated selectors if possible.

not possible

Reply:
Do the suggested selectors I provided not work?